pkg/ingester/wal_test.go

```diff
@@ -136,7 +136,9 @@ func TestCheckpointRepair(t *testing.T) {
 		w, ok = ing.wal.(*walWrapper)
 		require.True(t, ok)
 		// defer in case we hit an error though we explicitly close it later.
-		defer ing.Shutdown()
+		defer func() {
+			require.NoError(t, services.StopAndAwaitTerminated(context.Background(), ing))
+		}()
```

Comment:
```suggestion
		defer services.StopAndAwaitTerminated(context.Background(), ing)
```

I think this would suffice here, since we already check for NoError later in the test. This can only return error if test didn't get that far, and failed earlier.

Reply:
`StopAndAwaitTerminated` returns an error which would cause linter to fail. To make linter happy we would need a function there to throw away the error or add a `nolint` there. I feel whatever is there now looks more natural compared to `nolint` or throwing away the error.
What do you think?

Reply:
I'm OK keeping the current version. Let's merge it to fix `master`.

Reply:
Btw, we use `//nolint:errcheck` for these defers all over the tests.